### ISSUE TITLE
Chromecast support broken in v2.0.5, working in v2.0.2

### ISSUE THREAD

**Issue by molethiaz** [google/shaka-player#675]
Hi all,

i get an error in my custom receiver app. Maybe its a faulty configuration in Google Cast SDK Developer Console. So I receive no messages from custom receiver app to sender app.

- What version of Shaka Player are you using?
2.0.5

- Are you using the demo app or your own custom app?
Demo app adapted to own player. (Receiver-App / Shakaplayer)

- If custom app, can you reproduce the issue using our demo app?
Yes, I registred a receiver-app with source from: 'https://shaka-player-demo.appspot.com/demo/receiver_app.html'.
So i can debugging this application.

- What browser and OS are you using?
Mac 10.11.6 / Chromecast Firmware: 1.22.78337 / Google Chrome Version 56.0.2924.87
 
- What did you do?
1. I implemented Shaka-Demo-Player in my application. That's works well.
2. I tried Google Cast with demo receiverapp CC_APP_ID_ 658CCD53. That's works well, too.
3. I registered my custom receiver-app. Get demo files: receiver_app.css, receiver_app.html, receiver_app.js + controls.js, demo_utils.js + shaka player 2.0.5 compiled script.
4. I changed the CC_APP_ID_ to my custom receiver-app id. Videos are playing, but i get no messages (information about playback) from my castdevice / proxy. 
5. I tried a second custom receiver-app with hosted URL to 'http://shaka-player-demo.appspot.com/demo/receiver_app.html'
6. Same problem/error like the first custom receiver-app.

- What did you expect to happen?
Sender App (Google Chrome Browser) should show video information from receiver-app like currentTime, duration and i want to interact my video over sender app/browser over seekbar.

- What actually happened?
I don't get information about duration, currentTime from receiver-app to sender-app. So i cant control video on Chromecast except volume. I debugged receiver-app and get this error: 

***Compiled version:***
Exception: TypeError: this.a[b] is not a function at B.<anonymous>
b: "getPlayheadTimeAsDate"

***Source of error:*** https://github.com/google/shaka-player/blob/master/lib/cast/cast_receiver.js#L328

I think the events don't fire because of this error. (https://github.com/google/shaka-player/blob/87c7d5d6651fe4f6f03de84d77c920e91c34a107/lib/cast/cast_utils.js#L111)

Why can i interactive with the Shaka Demo receiver app (Id:658CCD53)? Maybe hosted URL isn't 'http://shaka-player-demo.appspot.com/demo/receiver_app.html'... I don't get it.

I hope someone can help me. 

Best regards,
 molethiaz

**Comment by molethiaz** [comment]
Hello again,

i just downgraded shaka-script in sender & receiver app to V 2.0.2 and now i get no error anymore. 

I saw method 'getPlayheadTimeAsDate' is not in PlayerGetterMethods Array: 
https://github.com/google/shaka-player/blob/v2.0.2/lib/cast/cast_utils.js#L105

A workaround may be:
```
shaka.cast.CastUtils.PlayerGetterMethods.forEach(function(name) {
    if(typeof (this.player_)[name] === 'function') {
        update['player'][name] = /** @type {Object} */(this.player_)[name]();
    }
 }.bind(this));
```

**Comment by joeyparrish** [comment]
Thank you for checking older versions and for reporting this issue.  We should be able to get it fixed in our next release.

**Comment by joeyparrish** [comment]
Thanks again for the report.  This was broken in v2.0.3 and will be fixed in v2.0.6.
